### PR TITLE
M2-6217: [Admin Panel] Clean up the cycle dependencies and enhance local performance

### PR DESCRIPTION
**Pull request by alex-i-e**
- [X] Related documentation has been added / updated
- [ ] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-6217](https://mindlogger.atlassian.net/browse/M2-6217)

### CODE REVIEW

File: config-overrides.js [hunk 1]
```diff
@@ -17,5 +20,29 @@ module.exports = function override(config) {
     }),
   );
 
+  if (process.argv.includes('checkCycles')) {
+    config.plugins.push(
+      new CircularDependencyPlugin({
+        exclude: /node_modules/,
+        onStart({ compilation }) {
```
Comment (IhorNesterenko99) [inline config-overrides.js]: Could you please resolve or disable these warnings?

File: src/shared/utils/index.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 export * from './colors';
 export * from './encryption';
 export * from './errors';
-export * from './renderComponentForEachTest';
-export * from './renderWithProviders';
+// export * from './renderComponentForEachTest';
```
Comment (alex-i-e) [inline src/shared/utils/index.ts]: will remove

File: src/modules/Builder/components/ConditionRow/Condition/Condition.test.tsx [hunk 3]
```diff
@@ -10,7 +10,8 @@ import {
   mockedSingleSelectFormValues,
   mockedSliderFormValues,
 } from 'shared/mock';
-import { getEntityKey, renderWithAppletFormData } from 'shared/utils';
+import { getEntityKey } from 'shared/utils';
+import { renderWithAppletFormData } from 'shared/utils/renderWithAppletFormData';
 import { ConditionRowType } from 'modules/Builder/types';
 import { ConditionType } from 'shared/consts';
```
Comment (IhorNesterenko99) [inline src/modules/Builder/components/ConditionRow/Condition/Condition.test.tsx:17]: Please disable the warning in this file.

Comment (alex-i-e) [reply src/modules/Builder/components/ConditionRow/Condition/Condition.test.tsx:17]: what do you mean? I propose to fix something different from this task in another thread, if it does not explicitly relate to the task

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**